tests/tests_disk_errors.yml

```diff
@@ -54,13 +80,40 @@
                 fs_type: 'ext3'
                 disks: "{{ unused_disks }}"
 
-        - name: Verify the output
+        - name: UNREACH
+          fail:
+            msg: "this should be unreachable"
+
+      rescue:
+        - name: Check that we failed in the role
           assert:
-            that: "{{ blivet_output.failed and
-                      blivet_output.msg|regex_search('cannot remove existing formatting on volume.*in safe mode') and
-                      not blivet_output.changed }}"
-            msg: "Unexpected behavior w/ existing data on specified disks"
+            that:
+              - ansible_failed_task.name != 'UNREACH'
+            msg: "Role has not failed when it should have"
+          vars:
+            # Ugh! needed to expand ansible_failed_task
```

Comment:
This comment seems to indicate that there is still something that needs to be done but to me it is unclear what is meant.

Reply:
@pcahyna can you answer this one?

Reply:
It is a workaround for a silly behavior in Ansible. The problem is, `ansible_failed_task.name` references the `storage_provider` variable which is not defined anymore at this point. But variables are evaluated lazily, so by evaluating `ansible_failed_task.name` here one crashes Ansible with an undefined variable reference. It should be done better, but I don't know how. (True, the comment should describe the problem better so it is not lost.)

Reply:
here are references to Ansible bug reports: https://github.com/ansible/ansible/issues/49942 https://github.com/ansible/ansible/issues/57399

Reply:
@tyll  For the record, #88 avoids the problem. https://github.com/linux-system-roles/network/pull/200 shows a different approach.